src/mavedb/lib/score_sets.py

```diff
@@ -62,22 +66,29 @@ def options(cls) -> list[str]:
         return [cls.NUCLEOTIDE, cls.TRANSCRIPT, cls.PROTEIN]
 
 
-def search_score_sets(db: Session, owner_or_contributor: Optional[User], search: ScoreSetsSearch) -> list[ScoreSet]:
+def search_score_sets(db: Session, owner_or_contributor: Optional["UserData"], search: ScoreSetsSearch) -> list[ScoreSet]:
```

Comment:
I think when we talked about this before what we had discussed was making the signature something like:
```
def search_score_sets(db: Session, owner_or_contributor: Optional["User"], search: ScoreSetsSearch, requesting_user: "UserData") -> list[ScoreSet]:
      ...
```

If we pass in the `owner_or_contributor` variable, we only fetch score sets which have that user as an owner or contributor. And then we use the `requesting_user` variable to check permissions. That way, we will be able to still search for other users' score sets but ensure that the user who requested them has permissions.

If we do it like that, I don't think we will need to make any changes to the existing search logic and don't need to add the `search.me` field.

Reply:
I suppose the alternative is to not check permissions or fetch superseding score set tails in the search function itself, but instead call the function that fetches tails and checks permissions after we return the score sets that match the search.

Reply:
Based on your suggestions, I did some modifications. The `search_score_set` function is nearly the same as before except removing filtering superseded score set out, and the permissions are checked outside of the search function. Let me know what you think about the new one. I prefer this way, but maybe miss some points.